scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -2044,23 +2226,28 @@ def run( self ):
         try:
             # check to see if the user has manually specified a cluster name, and if so then
             # force enable 'Starbuck' features
-            if self.__container_checker and self.__container_checker.k8s_cache.get_cluster_name() is not None:
+            if self.__container_checker and self.__container_checker.get_cluster_name( self.__container_checker.k8s_cache ) is not None:
                 self._logger.log( scalyr_logging.DEBUG_LEVEL_1, "Cluster name detected, enabling k8s metric reporting and controller information" )
                 self.__include_controller_info = True
                 self.__report_k8s_metrics = self.__report_container_metrics
 
+            # disable controller info if k8s queries are disabled
+            if self.__k8s_disable_api_server:
+                self.__include_controller_info = False
+
             if self.__report_k8s_metrics:
 
                 runtime = None
                 if k8s_cache:
+                    # TODO(czerwin):  If the api server has been disabled, we need to decide what to do here.
                     runtime = k8s_cache.get_container_runtime()
 
                 if runtime == 'docker':
                     self.__client = DockerClient( base_url=('unix:/%s'%self.__socket_file), version=self.__docker_api_version )
                     self.__metric_fetcher = DockerMetricFetcher(self.__client, self.__docker_max_parallel_stats )
 
                 k8s = KubernetesApi(k8s_api_url=self.__k8s_api_url)
-                self.__kubelet_api = KubeletApi( k8s )
+                self.__kubelet_api = KubeletApi( k8s, kubelet_api_host_ip=self._config.get('k8s_kubelet_host_ip') )
```

Comment:
kwarg should be `host_ip`, not `kubelet_api_host_ip`

Reply:
Good catch.. ok, we should fix.